pandas/io/excel/_xlsxwriter.py

```diff
@@ -212,7 +212,11 @@ def __init__(
             engine_kwargs=engine_kwargs,
         )
 
-        self._book = Workbook(self._handles.handle, **engine_kwargs)
+        try:
+            self._book = Workbook(self._handles.handle, **engine_kwargs)
+        except TypeError:
+            self._handles.handle.close()
+            raise
```

Comment:
Can you use `finally:` instead.

Reply:
@rhshadrach Sure. I'm not quite sure how you would want the try / finally statement to be set up. I modeled what I currently have based on the openpyxl implementation:
![image](https://github.com/pandas-dev/pandas/assets/45905457/aab11859-6945-43b5-8be1-5911e15d56a1)

Reply:
Is this what we're looking for? If so, do you want me to go ahead and update **_openpyxl.py** to follow the same structure?
![image](https://github.com/pandas-dev/pandas/assets/45905457/cd4cb0f1-418b-4d31-8a7a-181a36e8f6cd)

Reply:
In this case, the `except` portion is unnecessary. All you need is:

```
try:
    self._book = ...
finally:
    self._handles.handle.close()
```

Also, when sharing code I'd recommend using plain-text when possible rather than screenshots. It requires less  bandwidth, searchable, and copy-able.

Reply:
Ah, apologies - you can't use finally here. The finally portion is always called, which is not what we want (we only want to close when an exception is raised. Your original version was correct.